### PR TITLE
introduce backupProxy support

### PR DESCRIPTION
**Pull request by Alex-us**
1. Initialize UA with 2 servers:  outboundProxy and outboundProxyBackup
2.  outboundProxy  - is a main proxy.
3. outboundProxyBackup - is a backup proxy. Will be switched to, in case 2 situations: 
    a.  Socket disconnected and cannot be connected within maxReconnectionAttempts retries
    b. Received a message with status specified in regData.sipErrorCodes. If no values there, default are used : ['408', '502', '503']
4. Adding randomizing to interval for reconnections to avoid throttling : this.computeRandomTimeout
5. If connected to a backup, reconnecting to main proxy in following cases:
    a. If  `sipInfo.switchBackInterval`  is specified: switch after switchBackInterval + random (0, 15 min) to avoid server throttling (emit 
        'switchBackProxy' event to let client know we need to switch back to main);
    b. If `sipInfo.switchBackInterval` should be switched after next SIP provision update ( This should be done on client side)
6. Emit 'provisionUpdate' event in case `check-sync` message is arrived to notify client to update sip provision cached data;

### CODE REVIEW

**Comment by kirill-konshin** [comment]
Can you please reimplement this in `typescript` branch? It is going to be next release 0.7.0.

**Comment by Alex-us** [comment]
Sure, tomorrow will do

File: src/ringcentral-web-phone.js [hunk 1]
```diff
@@ -200,6 +214,120 @@
...
+              
+                if (switchBackInterval) {
+                    // Add random time to expand clients connections in time;
+                    switchBackInterval += this.computeRandomTimeout(1, 0, 15*60*1000);
+                    this.logger.warn('Try to switch back to main proxy after ' + Math.round(switchBackInterval/1000/60) + ' min');
+
+                    SIP.Timers.setTimeout(function(){
+                        this.configuration.wsServers[0].isError = false;
+                        //Notify client to trigger switching back to main proxy server;
+                        this.emit('switchBackProxy');
```
Comment (vyshakhbabji) [inline src/ringcentral-web-phone.js]: How should the client handle this? Should the client disconnect the active wsserver and then reconnect to the primary wsserver ?

File: src/ringcentral-web-phone.js [hunk 2]
```diff
@@ -126,20 +126,34 @@
...
-                        : this.sipInfo.wsServers,
+                wsServers: [outboundProxy, outboundProxyBackup],
                 traceSip: true,
-                maxReconnectionAttempts: options.maxReconnectionAttempts || 10,
-                reconnectionTimeout: options.reconnectionTimeout || 15,
-                connectionTimeout: options.connectionTimeout || 10,
-                keepAliveDebounce: options.keepAliveDebounce || 10
+                maxReconnectionAttempts: 3,
+                reconnectionTimeout: 3,
+                connectionTimeout: 5
```
Comment (vyshakhbabji) [inline src/ringcentral-web-phone.js]: is there a reason you decreased he timeouts and attempts ?

**Comment by kirill-konshin** [comment]
Superseeded by #198

**Pull request closed without merging**